File: longestConsecutiveSubse.cpp
class Solution {
public:
    // arr[] : the input array
    // N : size of the array arr[]

    //Function to return length of longest subsequence of consecutive integers.
    int findLongestConseqSubseq(int arr[], int N)
    {
        //Your code here
        unordered_map<int, int> lookup;
        int result = INT_MIN;
        int total = 0;
        for (int i = 0; i < N; i++) {
            lookup[arr[i]] = 1;
        }
        for (int i = 0; i < N; i++) {
            if (lookup.count(arr[i] - 1)) {
                continue;
            }
            else {
                int count = 0;
                int item = arr[i];
                while (true) {
                    if (lookup.count(item)) {
                        count++;
                    }
                    else {
                        break;
                    }
                    item++;
                }
                total += count;
                if (count > result) result = count;
            }
            if (total == N) {
                break;
            }
        }
        return result;

    }
};

File: longestConsecutiveSubse.py
class Solution:
    
    # arr[] : the input array
    # N : size of the array arr[]
    
    #Function to return length of longest subsequence of consecutive integers.
    def findLongestConseqSubseq(self,arr, N):
        #code here
        lookup = {}
        total = 0
        result = float('-inf')
        for i in range(N):
            if arr[i] not in lookup:
                lookup[arr[i]] = 1
            else:
                lookup[arr[i]] = 1
        for i in range(N):
            if arr[i] - 1 in lookup:
                continue
            else:
                count = 0
                item = arr[i]
                while True:
                    if item in lookup:
                        count += 1
                    else:
                         break
                    item += 1
                total += count
                if count > result:
                    result = count
            if total > N:
                break
        return result
